### PR TITLE
[Profiling] Consider static settings in status

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
Previously the profiling status check only considered the dynamically set value for `xpack.profiling.templates.enabled`. If a user set a value statically in `elasticsearch.yml` the status check did not reflect that leading to puzzling responses. With this commit we check not only dynamically defined cluster settings but also the static ones before returning the setting's default value.

### CODE REVIEW

**Comment by danielmitterdorfer** [comment]
Steps to reproduce:

1. Edit `config/elasticsearch.yml`:

```
xpack.profiling.templates.enabled: true
```

2. Start the cluster
3. Get profiling status: `GET /_profiling/status`, which shows:

```json
{
  "profiling" : {
    "enabled" : true
  },
  "resource_management" : {
    "enabled" : false
  },
  "resources" : {
    "created" : true
  }
}
```

whereas it should return `true` for `resource_management.enabled`.

Alternatively, this can be tested in development mode with by starting Elasticsearch as follows:

```
./gradlew :run -Drun.license_type=trial -Dtests.heap.size=4G -Dtests.es.xpack.profiling.templates.enabled=true --https
```

**Comment by danielmitterdorfer** [comment]
@rockdaboot I've assigned you as a reviewer. As I'm away in the next weeks it would be great if you could also merge this yourself after approval.

**Review by rockdaboot — APPROVED** [review]
LGTM

**Pull request merged**
